src/Simplex/Chat.hs

```diff
@@ -4489,6 +4517,21 @@ processAgentMessageConn user@User {userId} corrId agentConnId agentMessage = do
           groupMsgToView g' m ci msgMeta
         createGroupFeatureChangedItems user cd CIRcvGroupFeature g g'
 
+    xGrpDirectInv :: GroupInfo -> GroupMember -> ConnReqInvitation -> Maybe MsgContent -> RcvMessage -> MsgMeta -> m ()
+    xGrpDirectInv g m connReq mContent_ msg msgMeta = do
+      -- check direct messages preference
+      -- auto-join / client setting? (new to_join field if auto join is in UI?)
+      -- create contact (PendingMemberContact?)
+      -- if member already has contact, replace it (to not mix messages from connections)
```

Comment:
didn't get it

Reply:
If contact already exists for member receiving XGrpDirectInv (and doesn't exist for sending member), contact reference on member record will be replaced for the receiving member. Otherwise messages from two connections will be mixed in one contact. Or receiving member could prohibit XGrpDirectInv but then initiating member would never be able to connect directly